### PR TITLE
Fix importing error for reexported adapters

### LINKED ISSUE #1627: Installation instructions don't work
The documentation [encourages](https://next.material-ui-pickers.dev/getting-started/installation) the usage of:

```jsx
// pick a date util library
import DateFnsUtils from '@material-ui-pickers/adapter/date-fns';
```

Unfortunately, it doesn't seem to work: https://codesandbox.io/s/ecstatic-solomon-v4gvv.

## Environment

| Tech                 | Version |
| -------------------- | ------- |
| @material-ui/pickers | v4.0.0-alpha.5 |

### PR DESCRIPTION
**Pull request by dmtrKovalenko**
This PR closes #1627 

## Description
In order to make esm build work we need to provide separate `package.json` for each of adapter folder

### CODE REVIEW

File: lib/prepare-build-files.js [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    .copyFile(
+      path.resolve(__dirname, '..', 'README.md'),
+      path.resolve(__dirname, 'build', 'README.md')
+    )
+    .then(() => console.log('> Copied README.md'));
+}
+
+function createAdapterPackageFile(name) {
+  const packageJson = {
+    name: `@material-ui/pickers-adapter-${name}`,
```
Comment (dmtrKovalenko) [inline lib/prepare-build-files.js:15]: @oliviertassinari  What do you think will be the best name for adapter package.json? 
BTW `@material-ui/pickers/adapter/${name}` is not a valid name for package.json

Comment (oliviertassinari) [reply lib/prepare-build-files.js:15]: Do we even need to provide a name? For instance: https://unpkg.com/browse/@material-ui/core@4.9.9/Badge/package.json.

Comment (oliviertassinari) [reply lib/prepare-build-files.js:15]: We had https://github.com/mui-org/material-ui/issues/15715, but it seemed fine.

**Comment by oliviertassinari** [comment]
Should we update the documentation at the same time?

**Comment by dmtrKovalenko** [comment]
There is no breaking changes. Documentation was already updated — just not working

**Comment by oliviertassinari** [comment]
Note, [codesandbox-ci](https://codesandbox.io/docs/ci) could have helped to test the changes. We have configured it on the mono repository.

**Pull request merged**